app/src/main/java/kr/ac/kaist/team888/locator/Locator.java

```diff
@@ -192,14 +211,67 @@ private void applyCurve() {
     }
   }
 
-  private void setXrayPaths(Region canvasRegion) {
-    for (ArrayList<Stroke> skeleton : skeletons) {
+  private void applyContour() {
+    contours = new ArrayList<>();
+
+    for (ArrayList<Stroke> strokes: skeletons) {
+      // Stroke skeletons
+      ArrayList<BezierCurve> bezierCurves = new ArrayList<>();
+      for (Stroke stroke: strokes) {
+        ArrayList<Vector2D> points = new ArrayList<>();
+        points.add(new Vector2D(stroke.getStartPoint().getX(), stroke.getStartPoint().getY()));
+        ArrayList<Point2D> controlPoints = stroke.getControlPoints();
+        for (Point2D point : controlPoints) {
+          points.add(new Vector2D(point.getX(), point.getY()));
+        }
+        points.add(new Vector2D(stroke.getEndPoint().getX(), stroke.getEndPoint().getY()));
+
+        BezierCurve bezierCurve = new BezierCurve(points.toArray(new Vector2D[points.size()]));
+        if (bezierCurve.getOrder() <= 2
+            && bezierCurve.getPoint(0).equals(bezierCurve.getPoint(bezierCurve.getOrder()))) {
+          continue;
+        }
+        bezierCurves.add(bezierCurve);
+      }
+
+      // Convert Bezier curves to strokes
+      ArrayList<BezierCurve> contourBezier = BezierCurveUtils.stroke(bezierCurves, 30);
+      ArrayList<Stroke> contour = new ArrayList<>();
+      for (BezierCurve bezierCurve : contourBezier) {
+        Vector2D[] controlVectors = bezierCurve.getPoints();
+        Stroke.StrokeBuilder builder = new Stroke.StrokeBuilder();
+        builder.setStartPoint(new Point2D((float) controlVectors[0].getX(),
+            (float) controlVectors[0].getY()));
+        builder.setEndPoint(new Point2D((float) controlVectors[controlVectors.length - 1].getX(),
+            (float) controlVectors[controlVectors.length - 1].getY()));
+        for (int i = 1; i < controlVectors.length - 1; i++) {
+          builder.addControlPoint(new Point2D((float) controlVectors[i].getX(),
+              (float) controlVectors[i].getY()));
+        }
+        contour.add(builder.build());
+      }
+
+      contours.add(contour);
+    }
+  }
+
+  private void setPaths(Region canvasRegion, ArrayList<ArrayList<Stroke>> strokesSet,
```

Comment:
Because most of codes are sharable. Only different things are 1. saved list 2. existence of controlling points. I'll edit saved list into the parameter and rename `isSkeleton` to `showPoints`. Is this fix ok for you?